pypeclub/OpenPype
# General: Move loader logic from avalon to openpype

## Brief description
Move load workflow and related functions from avalon-core into OpenPype.

## Description
Moved bases of Loader plugin and load functions and also change names of functions and classes to identify their job. Load logic also consist of few database query functions that are loading related and at this moment don't have a place where they could be moved so they're part of load workflow. Added functions to register/deregister plugins/paths and to discover them for loading.

### Renamed variables
- `Loader` class renamed to `LoaderPlugin`
- `SubsetLoader` class renamed to `SubsetLoaderPlugin`
- `load` function renamed to `load_representation`
- `remove` function renamed to `remove_container`
- `update` function renamed to `update_container`
- `switch` function renamed to `switch_container`

## Note
This is not final structure of load workflow. The final structure rely on other structures and functions that are not yet moved and at the same time they can't be moved without this PR.

## Testing notes:
1. Run host
2. Show loader tool
3. Check if loader are available
4. Try to trigger load (at least one Loader)

### Tested hosts
- [x] Standalone Publisher

Related to PR https://github.com/pypeclub/avalon-core/pull/431
